validate/vendor: Disable pager for git diff

vvoland opened the pull request:
Prevent the failing validation from stalling the CI with:

```
FAIL: Vendoring was not performed correctly!
The following files changed during re-vendor:
M	vendor.mod
M	vendor.sum
--More--
```

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/moby/moby/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

thaJeztah commented on hack/validate/vendor:
```diff
@@ -45,11 +45,11 @@ else
 		echo 'The following files changed during re-vendor:'
 		echo
-		git diff --name-status HEAD -- "${vendor_files[@]}"
+		git --no-pager diff --name-status HEAD -- "${vendor_files[@]}"
```
Curious; was this always an issue, or did something change in either defaults, in GitHub actions, or because a TTY was attached (or not)?

Wondering if there's other `git` commands we run that may need the same treatment, and in that case, if we need to set a global option for this, e.g. setting `GIT_PAGER` in either GitHub actions or for the container; https://git-scm.com/docs/git/2.26.0#Documentation/git.txt-codeGITPAGERcode

vvoland replied:
I didn't see this happening before, so it looks like some GHA defaults changed.

vvoland committed:
Makefile: Pass PAGER/GIT_PAGER variable

Allow to override the PAGER/GIT_PAGER variables inside the container.
Use `cat` as pager when running in Github Actions (to avoid things like
`git diff` stalling the CI).

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

thaJeztah commented on Dockerfile:
```diff
@@ -18,6 +18,9 @@ ARG COMPOSE_VERSION=v2.24.5
 ARG SYSTEMD="false"
 ARG DOCKER_STATIC=1
 
+ARG PAGER
+ARG GIT_PAGER
+
```
Looks like we're not using these (we only set them at runtime through `-e`)

vvoland replied:
Oh right, we actually don't need these at build time.

thaJeztah reviewed (approved):
LGTM
